lineage/StateDistribution.py

```diff
@@ -0,0 +1,184 @@
+        unzipped_list_of_tuples_of_obs = list(zip(*list_of_tuples_of_obs))
+
+        # getting the observations as individual lists
+        # {
+        bern_obs = list(unzipped_list_of_tuples_of_obs[0])
+        exp_obs = list(unzipped_list_of_tuples_of_obs[1])
+        gamma_obs = list(unzipped_list_of_tuples_of_obs[2])
+
+        bern_p_estimate = bernoulli_estimator(bern_obs)
+        expon_scale_beta_estimate = exponential_estimator(exp_obs)
+        gamma_a_estimate, gamma_scale_estimate = gamma_estimator(gamma_obs)
+
+        state_estimate_obj = StateDistribution(state=self.state,
+                                               bern_p=bern_p_estimate,
+                                               expon_scale_beta=expon_scale_beta_estimate,
+                                               gamma_a=gamma_a_estimate,
+                                               gamma_scale=gamma_scale_estimate)
+        # } requires the user's attention.
+        # Note that we return an instance of the state distribution class, but now instantiated with the parameters
+        # from estimation. This is then stored in the original state distribution object which then gets updated
+        # if this function runs again.
+        return state_estimate_obj
+
+    def __repr__(self):
+        return "State object w/ parameters: {}, {}, {}, {}, {}.".format(self.state, self.bern_p, self.expon_scale_beta, self.gamma_a, self.gamma_scale)
+
+
+def prune_rule(cell):
+    """ User-defined function that checks whether a cell's subtree should be removed. """
+    truther = False
+    if cell.obs[0] == 0:
+        truther = True  # cell has died; subtree must be removed
+    return truther
+
+def tHMM_E_init(state):
+    return StateDistribution(state,
+                             0.9*(np.random.uniform()),
+                             50*(1+np.random.uniform()),
+                             7.5*(np.random.uniform()),
+                             1.5*(np.random.uniform()))
+
+# Because parameter estimation requires that estimators be written or imported, the user should be able to provide
+# estimators that can solve for the parameters that describe the distributions. We provide some estimators below as an example.
+# Their use in the ObservationEmission class is shown in the estimator class method. User must take care to define estimators that
+# can handle the case where the list of observations is empty.
+
+
+def report_time(cell):
+    """ Given any cell in the lineage, this function walks through the cell's ancestors and return how long it has taken so far. """
+    list_parents = [cell]
+    taus = [cell.obs[1]]
+
+    for cells in list_parents:
+        if cells._isRootParent():
+            taus.append(cells.obs[1])
+            break
+        elif cells.parent not in list_parents:
+            list_parents.append(cells.parent)
+            taus.append(cells.parent.obs[1])
+        summation = sum(taus)
+        return summation
+
+<<<<<<< HEAD
```

Comment:
delete ?

Reply:
I pushed my commits the other day! Seems like it hasn't been updated!!!